lib/registry.js

```diff
@@ -212,10 +205,10 @@ class Registry {
 	}
 }
 
-function formatLabels(labels) {
-	return Object.entries(labels).map(
-		([n, v]) => `${n}="${escapeLabelValue(v)}"`,
-	);
+function formatLabels(labels, filter) {
+	const entries = Object.entries(labels);
+	const filtered = filter ? entries.filter(filter) : entries;
+	return filtered.map(([n, v]) => `${n}="${escapeLabelValue(v)}"`);
```

Comment:
do we need to iterate the labels 3 times?

Reply:
Do you mean why `formatLabels` is called from three places? This is because:
1. for the shared labels, to generate the formatted string that is also saved in the map (line 221)
2. for the labels that aren't shared (line 61). This one includes a filter to prevent the shared labels to be included twice
3. for `exemplar.labelSet` (line 75)
All these iterations existed before. I consolidated them to use the same function instead of repeating the code.

Reply:
Simen was maybe asking--or if not, I'll ask--about the three iterations from `Object.entries`, `filter` and `map`. Might get some speed boost by using one plain loop.

Reply:
I refactored to use a single explicit loop (`for...of`) and it is indeed noticeably faster so I pushed this change.